### DISCUSSION TITLE
Cruiser 5.1.0 (Beta)

### DISCUSSION THREAD

**Discussion by devemux86** [devemux86/cruiser#671] [Announcements]
## Cruiser 5.1.0 (Beta) is [available](https://www.dropbox.com/scl/fo/0pi1svwz14rl49l2gldqr/ANVb1vCX3ezM-cuXr532fFY?rlkey=3lt3t50tinxdsdkfz0py1t4ec&st=e2snb8iu&dl=0):

- https://github.com/devemux86/cruiser/discussions/452#discussioncomment-11394482
(Settings | Navigation | Panels)

- https://github.com/devemux86/cruiser/discussions/667
(Settings | Application | Panels)

- https://github.com/devemux86/cruiser/discussions/660
(Settings | Application | Panels)

- https://github.com/devemux86/cruiser/discussions/204#discussioncomment-11284031
(see the link for instructions)

- https://github.com/devemux86/cruiser/discussions/622

- https://github.com/devemux86/cruiser/discussions/44#discussioncomment-11036929
(see the link for instructions)

- https://github.com/devemux86/cruiser/discussions/548

- https://github.com/devemux86/cruiser/discussions/644
(see the link for instructions)

- https://github.com/devemux86/cruiser/discussions/676#discussioncomment-11422001

- https://github.com/devemux86/cruiser/discussions/669 and voice navigation

- https://github.com/devemux86/cruiser/discussions/309#discussioncomment-11408233

- **Improved map rendering**

- **Many improvements**

<img src="https://github.com/user-attachments/assets/da05dc29-fc87-40e5-a235-9b0073cb7614" width="50%"/><img src="https://github.com/user-attachments/assets/ddb6ad2a-d113-485a-8c0e-ca7c7d1efb56" width="50%"/>
<img src="https://github.com/user-attachments/assets/58ee8e3e-f6f3-4215-8842-5d273f54faeb" width="50%"/><img src="https://github.com/user-attachments/assets/79876cef-995a-4c5e-9c11-adb3e3aaadbc" width="50%"/>

**Comment by bios4** [comment]
With all these new panel options and the fact, that winter is starting and "real navigation" is limited to (I assume) just a few users: do you see a chance to display all the panels configured to be shown during "Navigation" also for the simulation use case @devemux86?

Right now, after selecting "Simulate navigation" all panels are hidden except for these:

<img src="https://github.com/user-attachments/assets/605ecde4-e275-44d0-93a7-5694f7d5c57a" width="33%">

**Reply by devemux86** [reply]
As simulation is not real navigation, it uses virtual locations to explore the route.

Most panels (status, record, etc.) work with the GPS and all other sensors,
that are not used in simulation.

**Reply by devemux86** [reply]
Most features can also be tested with the "Walking" or other profile.
(not just driving)

**Reply by bios4** [reply]
This means "No", right? 😉

**Reply by devemux86** [reply]
Correctly, simulation is just a special view of navigation,
it does / can not have all its functions.

**Reply by Maverikk65** [reply]
@devemux86  I am planning to have vacation in January and February and to do some trips during that time. Might it be interesting that I install that beta and test? You know that I am not driving a motorbike, but a car... 
VG, NG.

**Reply by devemux86** [reply]
@Maverikk65 You can install the Beta over the current version.
It will be updated to the next version in the coming days.

**Comment by MotoUKRider** [comment]
Multi-route GPX (from BaseCamp) working nicely :man_dancing:

**Reply by devemux86** [reply]
@MotoUKRider Thanks for the feedback.

This is the new rewritten "**Import**" process, let's see what you all think of it.

**Comment by devemux86** [comment]
## Cruiser 5.1.0-b2 is [available](https://www.dropbox.com/scl/fo/0pi1svwz14rl49l2gldqr/ANVb1vCX3ezM-cuXr532fFY?rlkey=3lt3t50tinxdsdkfz0py1t4ec&st=e2snb8iu&dl=0):

- **GPS recording panel improvements**

- **Compass accuracy panel improvements**

**Comment by bios4** [comment]
Do I really have to hit "Record" now three times, when I start a route and e.g. want 1) GPS, 2) Trip A and 3) Trip B to start counting?

And also three times again if I want to stop?

Am I the only one who finds this super uncomfortable?

Or am I maybe the only one who hasn't understood how all these panels are supposed to work now!??

🤔

**Reply by devemux86** [reply]
(this is mentioned in the "Travel time" [announcement](https://github.com/devemux86/cruiser/discussions/670))

Auto recording of all panels in navigation / follow still exists, but is set separately from visibility:
- Visibility of all panels is set in "Settings | Application | Panels"

- Travel distance / time (auto) recording in navigation / follow is set in navigation panel settings

- GPS (auto) recording is set in GPS recording settings (like before)

**Reply by devemux86** [reply]
<img src="https://github.com/user-attachments/assets/d185af63-3286-4278-9eeb-154aa7b4b832" width="50%"/><img src="https://github.com/user-attachments/assets/5801e30b-22a4-4f63-b691-dc8fabd15b9f" width="50%"/>

**Comment by Leeb64** [comment]
When calculating routes online with GRAPHOPPER and Open Route Service, the following happens:

<img src="https://github.com/user-attachments/assets/b74a5246-b8df-40a9-a503-3cfa4a0b24c2" width="50%"/><img src="https://github.com/user-attachments/assets/bf27bedc-5467-44b5-a2cb-abb5ff944264" width="50%"/>

There are no problems with OSRM and Valhalla.
BRouter offline also works.

**Reply by devemux86** [reply]
@Leeb64 Thanks for the report.

OpenRouteService message mentions that it does not support very large no-go areas.
OSRM does not support no-go areas.

GraphHopper will be fixed in the next update.

**Reply by devemux86** [reply]
All active no-go areas exist in route requests.
It does not matter where the no-go areas are.
The router doesn't know until it starts calculating the route.

**Reply by Leeb64** [reply]
Perhaps you could think about automatically deactivating NGA for calculation programs that do not support NGA.

**Reply by devemux86** [reply]
Only CycleStreets and OSRM do not support no-go areas.
Their routing works fine without no-go areas.

**Reply by Ciutnik** [reply]
By the way, a quick question: shouldn't no-go areas be in the collection "Tracks / Areas"? Even by name it fits better here than to points (Favorites). I know it's a point with a radius, but... that's the definition of area 😃

**Reply by devemux86** [reply]
These are points like the favorites, so they belong to same category.
Don't be confused by the name. The name can change easily. :slightly_smiling_face: 

This should also cover most cases and calculates much faster the routes compared to no-go polygons.

**Reply by devemux86** [reply]
Maybe we could name the other category to "Tracks / Polygons".
(but names better be 1 word)

Or revert to its old name "Tracks".
A polygon is essentially a closed track.

**Reply by Ciutnik** [reply]
@devemux86 
I guess it all depends on the point of view. You look at it through the eyes of a programmer, I through the eyes of a regular user 😁 We don't need to change anything here.

Whatever you call it, the most important thing is that we have no-go and it works perfectly. Thanks for that 👍

**Comment by devemux86** [comment]
## Cruiser 5.1.0-b3 is [available](https://www.dropbox.com/scl/fo/0pi1svwz14rl49l2gldqr/ANVb1vCX3ezM-cuXr532fFY?rlkey=3lt3t50tinxdsdkfz0py1t4ec&st=e2snb8iu&dl=0):

- **GPS recording diagram displays speed**

- **Compass sensor setting was removed**

- Various improvements

**Reply by DolomitesTom** [reply]
Thanks a lot for bringing back the speed recording.

**Comment by ToToGSARider** [comment]


**Reply by kds112** [reply]
Why do you blame Cruiser (and Emux as well) again and again??? 2D view is NOT horrible. You made the proposal to re-introduce 3D. Okay, then please wait or enter the Mapsforge community in order to push the development. But please stop flooding this forum with your inappropriate opinion. Thanks.

Edit: Okay, you have deleted your post...

**Comment by DolomitesTom** [comment]
I can't find any accusations in GSArider's post. It's just an opinion that you can simply let stand in a democracy. You don't have to share it. I also sorely miss 3D in V5, but I still want development to continue. The good thing is that you can use both versions, V4 and V5 on different smart phones. I can live with that.

**Comment by devemux86** [comment]
## Cruiser 5.1.0-b4 is [available](https://www.dropbox.com/scl/fo/0pi1svwz14rl49l2gldqr/ANVb1vCX3ezM-cuXr532fFY?rlkey=3lt3t50tinxdsdkfz0py1t4ec&st=e2snb8iu&dl=0):

- **"Tracks / Areas" no-go flag is saved per profile**

- Various improvements

**Reply by Ciutnik** [reply]
The menu after tapping the distance/time panel in Polish should be different:
It is "Rekordowa/Zatrzymać" should be "Nagrywaj/Zatrzymaj". Although I think that instead of "Record" it would be more appropriate to use "Start" so as not to confuse it with recording a GPS track. "Start" fits the time panel even more.
 
If you decide to use the "Start/Stop" naming, you won't have to translate it into Polish, because it's the same 😁

**Reply by devemux86** [reply]
@Ciutnik Thanks for the translations.

Before changing any English texts, I want first to finish the [Chinese language](https://github.com/devemux86/cruiser/discussions/669) by @Ptckh.

**Comment by devemux86** [comment]
## Cruiser 5.1.0-b5 is [available](https://www.dropbox.com/scl/fo/0pi1svwz14rl49l2gldqr/ANVb1vCX3ezM-cuXr532fFY?rlkey=3lt3t50tinxdsdkfz0py1t4ec&st=e2snb8iu&dl=0):

- https://github.com/devemux86/cruiser/discussions/669 and voice navigation

**Comment by devemux86** [comment]
## Cruiser 5.1.0-b6 is [available](https://www.dropbox.com/scl/fo/0pi1svwz14rl49l2gldqr/ANVb1vCX3ezM-cuXr532fFY?rlkey=3lt3t50tinxdsdkfz0py1t4ec&st=e2snb8iu&dl=0):

- https://github.com/devemux86/cruiser/discussions/204#discussioncomment-11284031
(see the link for instructions)

**Comment by devemux86** [comment]
## Cruiser 5.1.0-b7 is [available](https://www.dropbox.com/scl/fo/0pi1svwz14rl49l2gldqr/ANVb1vCX3ezM-cuXr532fFY?rlkey=3lt3t50tinxdsdkfz0py1t4ec&st=e2snb8iu&dl=0):

- https://github.com/devemux86/cruiser/discussions/622#discussioncomment-11320050

- Travel panels revert to previous simple implementation with 2 distances and options in "Settings | Application | Panels".
As a new addition remains the manual start / stop actions in their menu available in all modes.

<img src="https://github.com/user-attachments/assets/cdfb128f-9bdb-49c7-823a-5d825be0f52c" width="50%"/>

**Reply by bios4** [reply]
The menu after long-pressing the "recording" panels now tells "Stop **GPS** recording" for both panels (distance AND time).
Are both panels really related to **GPS** recording, or should the time-related panel show a different text?

**Reply by bios4** [reply]
I could imagine to read "Stop distance recording" and "Stop time recording" instead...
While the icon doesn't show a "stop" but a "pause" function at all. 😁

**Reply by devemux86** [reply]
GPS recording menu is also available with tap (as there are not different actions on tap vs long press).
Stop icon is not nice to the eyes, just a large filled rectangle.

GPS recording is 1 process. It records GPS locations that have coordinates, time, elevation, accuracy, etc.
Both panels work together, they just display different information: recorded distance and recorded time.

**Reply by bios4** [reply]
Ah okay I see - thanks for explaining! 👍

**Reply by Ciutnik** [reply]
@bios4 
> doesn't show a "stop" but a "pause" function at all.

Because it is de facto a "pause" function. The next press will continue recording 🙂

@devemux86 
In Polish translation, it is better to use a different form of the word here. Instead of "Rozpocząć" please use "Rozpocznij".

**Reply by devemux86** [reply]
@Ciutnik Thanks, I will change the translation.

**Reply by kds112** [reply]
After installing this update the panel order is lost. It happened with some previous updates as well. Bug or feature? 😉

**Reply by devemux86** [reply]
@kds112 It is needed when we add / remove panels. :slightly_smiling_face:

**Reply by kds112** [reply]
I tried to reorder the panels but after saving nothing happened, the order stayed unchanged...?

**Reply by devemux86** [reply]
@kds112 Thanks, it will work properly in the next update.

**Comment by devemux86** [comment]
## Cruiser 5.1.0-b8 is [available](https://www.dropbox.com/scl/fo/0pi1svwz14rl49l2gldqr/ANVb1vCX3ezM-cuXr532fFY?rlkey=3lt3t50tinxdsdkfz0py1t4ec&st=e2snb8iu&dl=0):

- Various improvements

**Comment by devemux86** [comment]
## Cruiser 5.1.0-b9 is [available](https://www.dropbox.com/scl/fo/0pi1svwz14rl49l2gldqr/ANVb1vCX3ezM-cuXr532fFY?rlkey=3lt3t50tinxdsdkfz0py1t4ec&st=e2snb8iu&dl=0):

- Various improvements

**Comment by Ptckh** [comment]
Hi,

   What format does this HGT support, why do I use a 1 arc second .hgt file
made by myself, I can't recognize it

**Reply by devemux86** [reply]
@Ptckh Maybe it's easier to first test other hillshading data with desktop Cruiser via menu "Map | Open HGT".

**Reply by devemux86** [reply]
@Ptckh 

We use hillshading data 3" available on our Mapsforge download server.

3" data exist for the whole world and should be sufficient and fast for most cases.
1" data only exist for some countries and are slow to process and render the map.

---

To use other hgt data, you can clear the "hgt2" dir in Cruiser storage folder and put there your hgt files.

**Comment by Ptckh** [comment]
Okey, Thank you.

**Comment by devemux86** [comment]
## Cruiser 5.1.0-b10 is [available](https://www.dropbox.com/scl/fo/0pi1svwz14rl49l2gldqr/ANVb1vCX3ezM-cuXr532fFY?rlkey=3lt3t50tinxdsdkfz0py1t4ec&st=e2snb8iu&dl=0):

- Various improvements

**Comment by Ptckh** [comment]
Hi,   Is there any consideration for adding an external USB or Bluetooth
GNSS antenna for the windows version?

**Reply by devemux86** [reply]
@Ptckh Thanks for the suggestion. Do you mean on desktop Cruiser?

Desktop Cruiser is not intended for such use, better use Android Cruiser.

**Comment by Ptckh** [comment]
Yes, I found that I didn't find an app that could use Mapsforge Maps on the
desktop version. GARMIN basecamp can use GNSS on the desktop, but he can
only use Img.

**Comment by Leeb64** [comment]
I use CRUISER on 2 smartphones.
In Smartphone one, the height above sea level (165m) is displayed correctly.
On Smartphone "two", the height abovesea level is always displayed between 218 and 220 m.~55m too much.
A GPS test app shows the correct height on Smartphone"two".
Restarting the smartphone didn't change anything.
This morning I rode my bike and my smartphone("two") up a mountain. The mountain is 552m high. The GPS app also showed this - but in the CRUISER it was 598m. 46m too much...
Why does the GPS app show the correct value and CRUISER shows about 50m too much? - but only on  smartphone("two")...
1=DOOGEE S58pro // 2= SONY H8216    both Android 10

**Reply by devemux86** [reply]
The difference is the altitude above sea level or not. The altitude always comes from the GPS sensor.
OEMs may (or not) return the altitude above sea level, it depends on their custom implementations.

Android 14+ provides a global way to get the altitude above sea level on all devices.
We can not do custom things on old Androids, it is too much work and processing.

**Reply by Leeb64** [reply]
The whole thing isn't that important.... Nevertheless:
In earlier versions it worked properly.
The GPS app(GPS Test) gets it right too.
Maybe it would be possible to correct the height value manually...->+- (x )meters?

**Reply by devemux86** [reply]
Altitude correction is not the same on all locations and it is not known what OEMs return in their devices.
As everyone likes to modify Android and makes custom things. Altitude is fixed finally on latest Androids.

**Reply by MotoUKRider** [reply]
On Android 13, using both types of Location Service (including disabling WiFI/4G), Cruiser thinks I'm ~250ft ASL rather than ~70ft. The "GPS Test" app reports my correct altitude - this has GeoID compensation, so I guess that Cruiser isn't handling the readings in the same way.

**Reply by devemux86** [reply]
It requires too much extra processing that is not worth it on old Android.
When the latest versions offer it in a simple way globally for all devices.

**Comment by 0709wiwiwi** [comment]
Correct misplaced via points.
v 5.04. ok
v 5.1.0-b10. nok.

**Reply by devemux86** [reply]
@0709wiwiwi Thanks but this report is not complete. :slightly_smiling_face: 

We need a route example and the routing options or better the app profile.

---

Note: this function is handled by BRouter.
Without a detailed report we can not know if something and where is wrong.

We probably do nοt need it, users are responsible for placing their waypoints.
Automatic things from apps usually have negative results.

**Reply by 0709wiwiwi** [reply]
Questioning functions that worked fine before ? Never mind.

**Reply by devemux86** [reply]
I mentioned it in BRouter repository:
- https://github.com/abrensch/brouter/issues/387#issuecomment-2501009076

**Reply by bios4** [reply]
> Questioning functions that worked fine before ? Never mind.

Why this insulted reaction instead of - as Emux asked - providing an example route where the issue can be seen?

**Reply by devemux86** [reply]
Without route and app profile, we can not know if this is because of Cruiser or BRouter changes.

@afischerdev of BRouter mentioned that will check it, but it would be nice if we could help more.

**Reply by devemux86** [reply]
@afischerdev answered here:

- https://github.com/abrensch/brouter/issues/387#issuecomment-2501296251

Important: _"Note: this only works in the library. The web client collects data segment by segment."_

This means it requires calculation of the whole route, which is not something we want or do anymore for various reasons,
like use different profiles per route section, performance, etc.

So we should remove this setting, as this can no longer work under current conditions.
@0709wiwiwi Thanks for reporting this. :+1:

**Comment by MotoUKRider** [comment]
In beta-10, is there some change to the "snap to" process when starting a navigation?
When I begin, the map zooms in (to 18) but does not position itself at my location.  At first I thought it was a refresh issue, but it mostly depends where the map is at the time - often it's a white area, sometimes the sea and in a a crowded area you will see close-in stuff - but it doesn't initialize.

This is an existing observation noticed before the GPS/Altitude posts I made earlier.

**Reply by devemux86** [reply]
@MotoUKRider The only recent change was the compass filtering for testing.

What location service do you use?
Where is the location icon at that time?
What are the auto zoom settings?

Please provide the app profile, it is much easier to understand the reports.

If you can reproduce it, please provide a video to show what you see.

**Reply by devemux86** [reply]
@MotoUKRider I just made a quick test, it works correctly as always.

Please provide the app profile and a video to understand what you see.

**Reply by MotoUKRider** [reply]
@devemux86 I went as far back as beta-7, but the issue remains.

However, it appears Profile related: The default "Car" profile is fine, but my well-used "Motorcycle" and a night-mode profile exhibit this problem. My "Motorcycle" profile is attached - I'd rather not have to "design" the profile again from scratch so if there's a quick edit-fix I can apply then so much the better :smile_cat: 

I recall a similar old-vs-new Profile issue relating to BRouter a while back, and it forced and exit - You solved it within the App.

[Motorcycle.json.txt](https://github.com/user-attachments/files/17926050/Motorcycle.json.txt)

**Reply by MotoUKRider** [reply]
@devemux86 Interesting - I've fixed it!

I had set the Auto-Zoom to "Speed+Distance" a while back, and that's what is causing it. I have historically had it set to "Speed".  The former setting is not ideal for testing things sitting at your desk!

**Reply by devemux86** [reply]
@MotoUKRider Thanks for further looking into this! :+1: 

I can now reproduce it and will check it.

**Reply by devemux86** [reply]
@MotoUKRider Thanks for the report. :+1: 

It will work properly in the next update.

**Comment by devemux86** [comment]
## Cruiser 5.1.0-b11 is [available](https://www.dropbox.com/scl/fo/0pi1svwz14rl49l2gldqr/ANVb1vCX3ezM-cuXr532fFY?rlkey=3lt3t50tinxdsdkfz0py1t4ec&st=e2snb8iu&dl=0):

- https://github.com/devemux86/cruiser/discussions/452#discussioncomment-11394482
(Settings | Navigation | Panels)

- https://github.com/devemux86/cruiser/discussions/309#discussioncomment-11408233

- Various improvements

Display|Warning
-|-
![Screenshot_1732711219](https://github.com/user-attachments/assets/e71d3c07-1d4f-4e25-acde-03ac52e908dd)|![Screenshot_1732711237](https://github.com/user-attachments/assets/b265fc76-e536-421b-9bba-d0940bc8366f)

**Comment by troNpo** [comment]
Hello!! I'm developing a renderstheme, I would like to check if it works in Cruiser but when I go to the theme settings window the app closes,both the official version and in Beta . In Oruxmaps and Locus Maps it works Correct .I leave here the link to my blog where it can be downloaded, when possible please let me know what I need to correct.Thank you 
https://tronpoonpo.blogspot.com/p/exmapstema.html

**Reply by devemux86** [reply]
@troNpo Thanks for the report and the map theme!

Desktop Cruiser does not close, but does not open the layers screen.
I will also improve Android Cruiser to not close.

It is better to use an xml editor that has xml validation, it warns you when xml is not correct.

---

The map theme has an error, so this is the cause of the problems.

- Line 13: The `defaultvalue="EX_style_p"` does not exist, should be `EX_terreno` or `EX_supeposicion`

Also when xml has empty body, close it properly, instead of `<area fill="#0077CC"></area>` write `<area fill="#0077CC"/>`

**Reply by devemux86** [reply]
@troNpo If you run desktop Cruiser on the command line, it warns you about errors in the theme:
```
WARNING: Invalid style EX_style_p
```

---

You can run desktop Cruiser on the command line with:
```
java -jar cruiser.jar
```

**Reply by troNpo** [reply]
Thank you! Once the errors indicated have been corrected, it works correctly.Unfortunately my resources to develop themes are very limited, I only have an Android phone, a code editor and the map apps to verify the results.I use this website to format and detect errors in the xml https://appdevtools.com/xml-formatter
If you know of anything better for Android I would really appreciate it.Among other improvements to the theme I will try to include what you told me in the openandromaps forum.
https://www.openandromaps.org/en/oam-forums/topic/hillshading#post-56157
Which I sadly didn't see until yesterday. Thanks for the help.

**Reply by devemux86** [reply]
@troNpo About the hill shading in the layers menu, I reverted this change.

As no theme uses it, you can also avoid it so it is the same as other themes.

**Reply by troNpo** [reply]
Well it will be the best, Because the result is not the expected one, it does not work as a switch for the shading layer, what it does is amplify the magnitude of the existing hat.
https://youtube.com/shorts/Qyg8IN6ibsU?si=KhuLHtjip2WEuFi7
Thanks for the work anyway.
One last suggestion and I'll leave you alone :).It would be very helpful if Cruiser supported automatic installation of  theme.zip from android.
https://youtube.com/shorts/UayxPJF0ti8?si=mzpaDffbUlNnYMbt
Greetings, congratulations for the best and evolution of your app

**Comment by devemux86** [comment]
## Cruiser 5.1.0-b12 is [available](https://www.dropbox.com/scl/fo/0pi1svwz14rl49l2gldqr/ANVb1vCX3ezM-cuXr532fFY?rlkey=3lt3t50tinxdsdkfz0py1t4ec&st=e2snb8iu&dl=0):

- https://github.com/devemux86/cruiser/discussions/452#discussioncomment-11394482 improvements and display in simulation

- https://github.com/devemux86/cruiser/discussions/548

- https://github.com/devemux86/cruiser/discussions/676#discussioncomment-11422001

- Various improvements

**Reply by kds112** [reply]
What about a new release or new features to come first? 🙃

**Reply by devemux86** [reply]
@kds112 If everything is ok, a new release will probably be made soon.

**Comment by devemux86** [comment]
## Cruiser 5.1.0-b13 is [available](https://www.dropbox.com/scl/fo/0pi1svwz14rl49l2gldqr/ANVb1vCX3ezM-cuXr532fFY?rlkey=3lt3t50tinxdsdkfz0py1t4ec&st=e2snb8iu&dl=0):

- Various improvements

**Comment by devemux86** [comment]
## Cruiser 5.1.0-b14 is [available](https://www.dropbox.com/scl/fo/0pi1svwz14rl49l2gldqr/ANVb1vCX3ezM-cuXr532fFY?rlkey=3lt3t50tinxdsdkfz0py1t4ec&st=e2snb8iu&dl=0):

- https://github.com/devemux86/cruiser/discussions/678

- **[Minimum auto zoom can be set to 8](https://github.com/devemux86/cruiser/discussions/680#discussioncomment-11471309)**

- **Improved map rendering**

- Various improvements

**Reply by umpfmpf** [reply]
With 5.1.0-b14 while navigating the Poi's don't go off when i zoom out (Favorite is set to show starting at Zoom 10, but when i zoom out to less than 10 it still keeps Showing)
Without navigating it still works.

**Reply by devemux86** [reply]
@umpfmpf This has changed some versions ago.

Map elements during navigation are no longer auto-hide / re-appear for performance reasons.
Auto-hide of map elements remains a feature during route planning and browse the map.

**Reply by bios4** [reply]
> Map gestures in navigation

Is this limited to finger/pinch **_zoom_** only?
Or is it a bug, that map **_rotation_** with fingers is (still) not possible during navigation?

**Reply by devemux86** [reply]
@bios4 

If "Movement direction" is selected as map orientation, then GPS is the only one responsible for the map rotation.
Otherwise there can be problems with 2 rotations at the same time.

**Reply by umpfmpf** [reply]
This means that if I zoom out while driving, all the poi's will remain visible, no matter what zoom level?

![Screenshot_20241206_110432_Cruiser](https://github.com/user-attachments/assets/47918f95-fcc7-4e7f-aec1-927818eb4794)

**Reply by Maverikk65** [reply]
Wow! Many thanks. Emus @devemux86 you are genius! I installed the new version right now and will drive 450km later today in the evening... (mostly autobahn of course, but also 1 hour on the roads...). 

For me it is very good that it does not rotate --- I already had some situations where I got disoriented in Copilot when I zoomed out to much and a little bump gave me an additional rotation and I got totally lost... ;-) Thankfully I am not flying, only driving...

**Reply by devemux86** [reply]
@umpfmpf We must give priority first to stability and performance.

If want to examine the map, **pause** navigation and everything works as before.
While navigation works, we are not supposed to play with the map.

**Reply by umpfmpf** [reply]
@0709wiwiwi 
I use only Petrol from Aral or Shell because I need more than 100 Octane. So I have them as Favorites.

**Reply by devemux86** [reply]
@umpfmpf Displaying a large number of favorites could be useful in route planning when browsing the map.

To use nearby favorites in navigation (like fuel stations), it is better to hide them and select "Use in navigation".
Then only the favorites near the route will be displayed on the map and in the nearby panels.

**Comment by kds112** [comment]
Will you release a 5.1 version or will you go directly to 5.2 beta? 5.2 was already announced several times...

**Reply by devemux86** [reply]
Development of new features does not stop! :slightly_smiling_face: 

**5.1** will be released, then **5.2** will be Beta for testing.

**Comment by devemux86** [comment]
## Cruiser 5.1.0-b15 is [available](https://www.dropbox.com/scl/fo/0pi1svwz14rl49l2gldqr/ANVb1vCX3ezM-cuXr532fFY?rlkey=3lt3t50tinxdsdkfz0py1t4ec&st=e2snb8iu&dl=0):

- Various improvements

**Comment by Ptckh** [comment]
hi,


[image: da9217bbee1fc0ed778b809abad7fd2c.png]

**Reply by devemux86** [reply]
@Ptckh Please use the browser to attach screenshots.

**Comment by Ptckh** [comment]
Ok, See the attachment please.

**Reply by devemux86** [reply]
@Ptckh There is no attachment.
Use the browser and the "Attach files" button of GitHub, not e-mails.

Also please write details about what you report.

**Comment by Ptckh** [comment]
There's a garbled code.
![7fd2c](https://github.com/user-attachments/assets/9590c18b-d26b-4469-8459-9e078b09827d)

**Reply by devemux86** [reply]
@Ptckh It works here on Linux with OpenJDK 17.

What Java do you use? Try OpenJDK from [here](https://adoptium.net/).

**Comment by Ptckh** [comment]
Ok, I configured JDK8 and now I'm switching to JDK21

**Comment by Ciutnik** [comment]
When planning bicycle routes (built-in bike profile), I use BRouter offline, but there are no alternatives offered. Was it always like this? I have it set to 2 alternatives.

**Reply by devemux86** [reply]
@Ciutnik Thanks for the report. It seems to happen with driving profiles too?

**Reply by Ciutnik** [reply]
It really is so! I didn't think to check this on other profiles... 🙂

**Reply by devemux86** [reply]
@Ciutnik This could be because of the latest BRouter code. I informed @afischerdev here:
- https://github.com/abrensch/brouter/issues/387#issuecomment-2535092076

**Reply by Ciutnik** [reply]
@devemux86 Thanks for reporting the issue to the BRouter developers. I hope they fix it quickly.

**Reply by devemux86** [reply]
This is fixed by @afischerdev in BRouter and will be included in the next Cruiser update.

@Ciutnik Thanks again for the report! :+1:

**Reply by Ciutnik** [reply]
I would like to thank both of you gentlemen @devemux86 & @afischerdev for the quick fix 🙂👍

**Comment by devemux86** [comment]
## Cruiser 5.1.0-b16 is [available](https://www.dropbox.com/scl/fo/0pi1svwz14rl49l2gldqr/ANVb1vCX3ezM-cuXr532fFY?rlkey=3lt3t50tinxdsdkfz0py1t4ec&st=e2snb8iu&dl=0):

- **BRouter (offline) alternative routes fix**

- Various improvements

**Comment by Ptckh** [comment]
Hi, I'm using cruiser,OSRM to intelligently plan a route and I'm finding that there is a lack of elevation information so that there is no way to calculate the cumulative ascent and descent, is it possible to add the elevation? Thanks.


---- Replied Message ----
| From | ***@***.***> |
| Date | 12/13/2024 18:55 |
| To | ***@***.***> |
| Cc | ***@***.***>,
***@***.***> |
| Subject | Re: [devemux86/cruiser] Cruiser 5.1.0 (Beta) (Discussion #671) |

Closed #671 as resolved.

**Reply by devemux86** [reply]
@Ptckh OSRM does not provide elevation information in the calculated routes.

Better use another router, like **BRouter (offline)** that has many more options.

---

**Cruiser 5.2.0** is on [Google Play](https://play.google.com/store/apps/details?id=gr.talent.cruiser):
- https://github.com/devemux86/cruiser/discussions/701

**Comment by Ptckh** [comment]
Ok, thank you.
---- Replied Message ----
| From | ***@***.***> |
| Date | 1/12/2025 16:43 |
| To | ***@***.***> |
| Cc | ***@***.***>,
***@***.***> |
| Subject | Re: [devemux86/cruiser] Cruiser 5.1.0 (Beta) (Discussion #671) |

@Ptckh OSRM does not provide elevation information in the calculated routes.

Better use another router, like BRouter (offline) that has many more options.

Cruiser 5.2.0 is on Google Play:

#701
